### DISCUSSION TITLE
Choose image from file library - filter image file types only

### DISCUSSION THREAD

**Discussion by nickluger** [directus/directus#6097] [Feature Requests]
When selecting an existing image from the file library, only image file types like `png` or `jpg`  should be selectable per default, or there should be at least a way to filter out other files in the drawer. We have many file types in our library, like for example `PDF` or `.html` files, and they are all shown in the selector/chooser when picking an existing file from the lib, which causes unnecessary noise for the editor.

Currently, there is only a search field and a sort field, but no way to filter things.

What would be the preferred way to handle this issue? 

![Bildschirmfoto 2021-06-05 um 19 43 52](https://user-images.githubusercontent.com/32647249/120900674-6cc02f00-c636-11eb-983d-cdb1517c3db6.png)

**Comment by benhaynes** [comment]
We'd need to figure out how this will get injected onto the page. Not sure about adding the sidebar into the drawer. We _could_ look into bringing _Advanced Filter_ back as a dropdown of the search bar... but that's a bigger discussion.

**Comment by rijkvanzanten** [comment]
> What would be the preferred way to handle this issue?

I think we'll have to look into upgrading the API first to handle "select existing" slightly different. Currently, the app simply sends up the UUID of the selection, which negates the use of nested filter permissions. Take the following example:

```
{
  "author": {
    "name": {
      "_eq": "Rijk"
    }
  }
}
```

When I create a new Item as follows, it validates as expected:

```
{
  "author": {
    "name": "Nick"
  }
}
```

However, when I use an existing item, the following is the payload:

```
{
  "author": "863f0f6e-ca4f-49ac-9c99-f3d9666c974a"
}
```

in which case, the `name` field isn't set, and thus not validated.

We should upgrade the API to dynamically read in the existing item in the nested relational validation step, so the same permission rule from the parent item is checked on the nested primary key of the child. That way, we can rely on the following for the file permissions:

```
{
  "file": {
    "type": {
      "_in": ["image/png", "image/jpeg"]
    }
  }
}
```

**Reply by nickluger** [reply]
OK, wouldn't have thought, this touches Directus on such a deep level. I assume this should be tackled by the core team, then?

Also: Should image types be set as a *default* filter on the image selector?

**Reply by rijkvanzanten** [reply]
It's also a question whether or not this has to be on the permissions level in the API, or "just" an added filter for the app 🤔

**Reply by nickluger** [reply]
Thinking about it 🤔  - it could be beneficial (in general) to be able to filter the possible relational items, beyond this specific image use case. 

Just as a real life example: In an e-commerce use-case I'd have a `T-Shirt: Red` which would be a variant of `T-Shirt: Generic`. Thinking there could be millions of different products, but I want to make sure that T-shirts can only be variants of other T-shirts, while there are hundreds of other product types in the `products`-collection like bags or shoes. So when selecting the self-relation, I'd like to be able to filter out other product types. 

Of course I can guarantee this business rule is adhered to, via hook, but from the editor's perspective it might be beneficial. So it's more of an app thing, IMHO.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
